### PR TITLE
[GCP] Auto update gcp catalog

### PR DESCRIPTION
**Pull request by Michaelvll**
This automatically update the GCP catalog for the user, so that they can get the latest price.

Related to: https://github.com/skypilot-org/skypilot-catalog/pull/36

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/service_catalog/data_fetchers/fetch_gcp.py [hunk 1]
```diff
@@ -138,7 +140,7 @@ def _get_unit_price(sku: Dict[str, Any]) -> float:
     return units + nanos
 
 
-@functools.lru_cache(maxsize=None)
+@functools.lru_cache(maxsize=1)
```
Comment (WoosukKwon) [inline sky/clouds/service_catalog/data_fetchers/fetch_gcp.py:143]: Just curious: what's the reason for this change?

File: sky/clouds/service_catalog/gcp_catalog.py [hunk 2]
```diff
@@ -16,8 +16,15 @@
 if typing.TYPE_CHECKING:
     from sky.clouds import cloud
 
-_df = common.read_catalog('gcp/vms.csv')
-_image_df = common.read_catalog('gcp/images.csv')
+# Pull the latest catalog every 2 weeks.
+# GCP guarantees that the catalog is updated at most once per 30 days, so
+# we do not need to pull too frequently.
+_PULL_FREQUENCY_HOURS = 24 * 14
```
Comment (WoosukKwon) [inline sky/clouds/service_catalog/gcp_catalog.py:22]: In the worst case the user's catalog will be updated 14 days after the actual change is made, right? Why don't we decrease this a bit (e.g., 1 week or so)? I think that will do more good than harm.

**Review by WoosukKwon — APPROVED** [review]
LGTM. Left minor comments.

File: sky/clouds/service_catalog/data_fetchers/fetch_gcp.py (hunk 1 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_gcp.py]: Ahh, it is a remnant. Changed it back. Thanks!

File: sky/clouds/service_catalog/gcp_catalog.py (hunk 2 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/gcp_catalog.py]: Yes, that is a good point. I was afraid that the fetch_gcp is not robust enough, causing our catalog somehow corrupted, but it should be fine to set it to 7 days, and we can fix any issues for that if it ever happens.

**Comment by Michaelvll** [comment]
Thanks @WoosukKwon for the review! Merging. ; )
